composer/utils/checkpoint.py

```diff
@@ -354,11 +356,14 @@ def format(self, state: State, is_deepspeed: bool = False, keep_placeholders: bo
 
 
 def is_checkpoint_legacy_sharded(object_store: Optional[ObjectStore], source_path: str):
+    if source_path.endswith('.symlink') or os.path.islink(source_path):
+        source_path = extract_path_from_symlink(source_path, object_store=object_store)
     metadata_path = str(Path(source_path) / Path('.metadata'))
     if object_store is None:
         return not os.path.exists(metadata_path)
     else:
         try:
+            _, _, metadata_path = parse_uri(metadata_path)
```

Comment:
why is this change needed? Is it because parent calling fn does this but if we read a symlink, we need to redo this? If so, can we do this in the symlink if statement?

Reply:
good call. fixed

Reply:
if it's not a symlink it will still have the prefix in it. calling this `_, _, metadata_path = parse_uri(metadata_path)` is like pressing lock on your car key fob. Useful if your car is unlocked and has no effect if you car is already locked, so we should just do it